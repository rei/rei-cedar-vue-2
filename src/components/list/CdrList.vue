<template>
  <component
    :is="tag"
    :class="modifierClass"
  >
    <slot />
  </component>
</template>

<script>
import modifier from 'mixinsdir/modifier';

export default {
  name: 'CdrList',
  mixins: [modifier],
  props: {
    /**
     * Tag accepts a user defined element and expects either: 'ul' = Unordered List or 'ol' = Ordered List.
     */
    tag: {
      type: String,
      default: 'ul',
      validator: value => ['ul', 'ol'].indexOf(value) >= 0 || false,
    },
  },
  computed: {
    baseClass() {
      return 'cdr-list';
    },
  },
};
</script>

<style module>
  @import 'cssdir/settings/_index.pcss';
  @import './styles/CdrList.pcss';
</style>
