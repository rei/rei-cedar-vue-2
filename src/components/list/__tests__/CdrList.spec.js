import { mount } from '@vue/test-utils';
import CdrList from 'componentdir/list/CdrList';

describe('CdrList', () => {
  test('renders correctly', () => {
    const wrapper = mount(CdrList);
    expect(wrapper.element).toMatchSnapshot();
  });
  
  it('renders an ol', () => {
    const wrapper = mount(CdrList, {
      propsData: {
        tag: 'ol',
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});