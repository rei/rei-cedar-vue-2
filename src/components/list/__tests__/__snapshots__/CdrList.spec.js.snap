// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrList renders an ol 1`] = `
<ol
  class="cdr-list cdr-list "
/>
`;

exports[`CdrList renders correctly 1`] = `
<ul
  class="cdr-list cdr-list "
/>
`;
